src/cfnlint/schema_manager.py

```diff
@@ -0,0 +1,87 @@
+"""
+Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: MIT-0
+"""
+
+import json
+import logging
+import os
+
+import botocore.exceptions
+import boto3
+from cfnlint.template import Template
+
+LOGGER = logging.getLogger(__name__)
+
+
+class SchemaManager(object):
+    """Download schemas if necessary and validate modules"""
+    def __init__(
+            self, filename, template, regions):
+        self.filename = filename
+        self.regions = regions
+        self.template = Template(filename, template, regions)
+        self.boto3_sts = boto3.client('sts')
+        self.boto3_cfn = boto3.client('cloudformation')
+
+    # Check if the appropriate folder already exists
+    def check_folders(self, name, registry_type):
+        account_id = self.boto3_sts.get_caller_identity().get('Account')
+        for region in self.regions:
+            path = self.create_path(account_id, region, name)
```

Comment:
It seems like we pass the path around a lot.  Is there a better way to do this?  Like can we set the base path once in the init `base_path = os.path.join(os.path.expanduser('~'), '.cloudformation', account_id)`